### PR TITLE
Implement "modin.pandas.show_versions()" and "python -m modin --versions"

### LINKED ISSUE #3971: modin.show_versions similar to pandas
pandas provides a concise but effective way of gathering the information of core packages: [pandas.show_versions()](https://pandas.pydata.org/pandas-docs/dev/reference/api/pandas.show_versions.html):
```
>>> pd.show_versions()

INSTALLED VERSIONS
------------------
commit           : 945c9ed766a61c7d2c0a7cbb251b6edebf9cb7d5
python           : 3.8.8.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : Intel64 Family 6 Model 165 Stepping 2, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : Russian_Russia.1251

pandas           : 1.3.4
numpy            : 1.21.3
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 21.3.1
setuptools       : 52.0.0.post20210125
Cython           : None
pytest           : 6.2.5
hypothesis       : None
sphinx           : 3.5.3
blosc            : None
feather          : 0.4.1
xlsxwriter       : None
lxml.etree       : 4.6.3
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.3
IPython          : 7.29.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : 1.3.2
fsspec           : 0.8.7
fastparquet      : None
gcsfs            : None
matplotlib       : 3.2.2
numexpr          : 2.7.3
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : 0.15.0
pyarrow          : 1.0.1
pyxlsb           : None
s3fs             : 0.4.2
scipy            : 1.7.1
sqlalchemy       : 1.4.26
tables           : 3.6.1
tabulate         : None
xarray           : 0.19.0
xlrd             : 2.0.1
xlwt             : None
numba            : None
```

It would be good for our users to have the same mechanism which would augment pandas function with the packages Modin uses.

I also propose to make it callable from the command line via `python -m modin --versions`

### PR DESCRIPTION
**Pull request by vnlitvinov**
## What do these changes do?
This is an augmentation for [pandas.show_versions()](https://pandas.pydata.org/pandas-docs/dev/reference/api/pandas.show_versions.html) plus an ability to call this from command line in an easy way.

<details><summary>Example output</summary>

```
>python -m modin --versions

INSTALLED VERSIONS
------------------
commit           : 66e3805b8cabe977f40c05259cc3fcf7ead5687d
python           : 3.8.8.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : Intel64 Family 6 Model 165 Stepping 2, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : Russian_Russia.1251

Modin dependencies
------------------
modin            : 0.12.0+80.gdd2c53f6
ray              : 1.6.0
dask             : 2021.10.0
distributed      : 2021.10.0
omniscidbe       : None

pandas dependencies
-------------------
pandas           : 1.3.5
numpy            : 1.21.3
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 21.3.1
setuptools       : 52.0.0.post20210125
Cython           : None
pytest           : 6.2.5
hypothesis       : None
sphinx           : 3.5.3
blosc            : None
feather          : 0.4.1
xlsxwriter       : None
lxml.etree       : 4.6.3
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.3
IPython          : 7.29.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : 1.3.2
fsspec           : 2022.01.0
fastparquet      : None
gcsfs            : None
matplotlib       : 3.2.2
numexpr          : 2.7.3
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : 0.15.0
pyarrow          : 1.0.1
pyxlsb           : None
s3fs             : 2022.01.0
scipy            : 1.7.1
sqlalchemy       : 1.4.26
tables           : 3.6.1
tabulate         : None
xarray           : 0.19.0
xlrd             : 2.0.1
xlwt             : None
numba            : None
```
</details>

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/developer/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3971
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/utils.py [hunk 1]
```diff
@@ -561,3 +572,103 @@ def import_optional_dependency(name, message):
...
+            )
+
+    try:
+        # We import ``PyDbEngine`` from this module since correct import of ``PyDbEngine`` itself
+        # from Omnisci is located in it with all the necessary options for dlopen.
+        from modin.experimental.core.execution.native.implementations.omnisci_on_native.utils import (  # noqa
+            PyDbEngine,
+        )
+
+        result["omniscidbe"] = "present"
```
Comment (YarShev) [inline modin/utils.py:613]: Is there a way to know omniscidbe version?

File: modin/utils.py [hunk 2]
```diff
@@ -561,3 +572,103 @@ def import_optional_dependency(name, message):
...
+    as_json : str or bool, default: False
+        * If False, outputs info in a human readable form to the console.
+        * If str, it will be considered as a path to a file.
+          Info will be written to that file in JSON format.
+        * If True, outputs info in JSON format to the console.
+
+    Notes
+    -----
+    This is mostly a copy of pandas.show_versions() but adds separate listing
+    of Modin-specific dependencies
```
Comment (YarShev) [inline modin/utils.py:640]:
```suggestion
    of Modin-specific dependencies.
```

File: modin/utils.py [hunk 3]
```diff
@@ -561,3 +572,103 @@ def import_optional_dependency(name, message):
...
+    """
+    sys_info = _get_sys_info()
```
Comment (YarShev) [inline modin/utils.py:642]: Which commit will be shown from commit history? And also, will it be from pandas or from Modin?

File: modin/utils.py [hunk 4]
```diff
@@ -561,3 +572,103 @@ def import_optional_dependency(name, message):
...
+        else:
+            assert isinstance(as_json, str)  # needed for mypy
+            with codecs.open(as_json, "wb", encoding="utf8") as f:
+                json.dump(j, f, indent=2)
+
+    else:
+        assert isinstance(sys_info["LOCALE"], dict)  # needed for mypy
+        language_code = sys_info["LOCALE"]["language-code"]
+        encoding = sys_info["LOCALE"]["encoding"]
+        sys_info["LOCALE"] = f"{language_code}.{encoding}"
```
Comment (YarShev) [inline modin/utils.py:664]: Why do we need to override this value?

File: modin/utils.py (hunk 1 above)
Comment (vnlitvinov) [reply modin/utils.py:613]: I didn't find any (except querying by something like `conda list`). Maybe there's some C API of sorts... @modin-project/modin-omnisci ideas?

File: modin/utils.py (hunk 3 above)
Comment (vnlitvinov) [reply modin/utils.py:642]: Good question, it's a commit from pandas. I've replaced it with Modin info, thanks for the catch.

File: modin/utils.py (hunk 4 above)
Comment (vnlitvinov) [reply modin/utils.py:664]: This is mostly a copy-paste from `pandas.show_versions()`, I don't know why they did it so 🙃

**Review by devin-petersohn — COMMENTED** [review]
Can we just instead do `from pandas import show_versions`?

**Comment by vnlitvinov** [comment]
Yes and no. I wanted to integrate Modin components smoothly, which is impossible without effectively re-writing the function.
It's also possible to extract the output from `pandas.show_versions()` and re-compose it ourselves, but this function can only output to a file pointed by _name_, which makes a cumbersome code to get it programmatically to just re-create.

I think this code would be changed very rarely when it stabilizes in our repo.

**Comment by devin-petersohn** [comment]
>but this function can only output to a file pointed by name, which makes a cumbersome code to get it programmatically to just re-create.

Can we not do this:

```python
def show_versions(as_json):
    from pandas import show_versions as _show_versions
    
    if not as_json:
        _show_versions()
        # Modin logic here
    else:
        # capture stdout and add json logic
```

**Comment by vnlitvinov** [comment]
I've inserted Modin components in between, so I'd have to capture the stdout anyway to achieve the same thing.

I'd have to duplicate the logic of composing the output after that anyway.

So what's I'm gaining is only _not_ importing and using some protected functions from pandas, which is a fair price so that I don't have to muck with grabbing the output.

**Review by devin-petersohn — COMMENTED** [review]
Can we add this file to codecov ignores?

Otherwise, I am ok with these changes.

File: modin/utils.py (hunk 1 above)
Comment (ienkovich) [reply modin/utils.py:614]: Currently, we don't have any API to get omniscidbe version

Comment (YarShev) [reply modin/utils.py:614]: Does  `conda list` way make sense to use?

**Comment by vnlitvinov** [comment]
@devin-petersohn I've excluded `__main__.py`, could we merge this PR before a release?

**Comment by devin-petersohn** [comment]
@vnlitvinov can we also remove the new functionality from codecov, or do you plan to add a test later?

File: modin/utils.py (hunk 1 above)
Comment (vnlitvinov) [reply modin/utils.py:614]: Too many assumptions (and `conda list` is a slow thing to me).
I would rather we add an API function to get the version.

**Comment by vnlitvinov** [comment]
I've added a small test already: https://github.com/modin-project/modin/pull/4007/files#diff-003a411f9eba9c58275576d3bfeddd0815d72660fce91c34f77052d6fe5b92a5R257-R265

**Review by devin-petersohn — APPROVED** [review]
LGTM. Codecov is complaining about some lines, but functionality looks good.

**Review by YarShev — APPROVED** [review]

File: modin/utils.py (hunk 1 above)
Comment (gshimansky) [reply modin/utils.py:614]: Also we assume that `conda` is present in the PATH which is not guaranteed.

**Pull request merged**